phpBB/includes/acp/acp_main.php

```diff
@@ -646,11 +646,10 @@ function main($id, $mode)
 		}
 
 		// Warn if no search index is created
-		if ($config['num_posts'] && class_exists($config['search_type']))
+		if ($config['num_posts'])
 		{
-			$error = false;
-			$search_type = $config['search_type'];
-			$search = new $search_type($error, $phpbb_root_path, $phpEx, $auth, $config, $db, $user, $phpbb_dispatcher);
+			$search_backend_factory = $phpbb_container->get('search.backend_factory');
+			$search = $search_backend_factory->get_active();
```

Comment:
What happens if the defined service is not available for any reason? Shouldn't we catch the exception and handle this similarly like we did before with `class_exists()`?

Reply:
The service class should exist, except if a extension with an active search backend is removed, but the extension should handle that i think.
I can launch an excepcion from the factory class if you want

Reply:
Here if the service don't exist, a RuntimeException with the message "No service found..." will be thrown

Reply:
Now it triggers an error saying search backend not found